syncd/FlexCounter.cpp

```diff
@@ -15,3098 +15,1129 @@ using namespace std;
-                if (m_statsMode == SAI_STATS_MODE_READ_AND_CLEAR && !isStatsModeSupported(currentStatModes, SAI_STATS_MODE_READ_AND_CLEAR))
-                {
-                    continue;
-                }
-                sai_ingress_priority_group_stat_t counter = static_cast<sai_ingress_priority_group_stat_t>(statCapability.stat_enum);
-                m_supportedPriorityGroupCounters.insert(counter);
-            }
-        }
+    SWSS_LOG_ENTER();
+
+    for (const auto &kv : m_counterContext)
+    {
+        kv.second->removePlugins();
     }
-    return status;
+
+    m_isDiscarded = true;
 }
 
-void FlexCounter::getSupportedPriorityGroupCounters(
-        _In_ sai_object_id_t priorityGroupRid,
-        _In_ const std::vector<sai_ingress_priority_group_stat_t> &counterIds)
+void FlexCounter::addCounterPlugin(
+        _In_ const std::vector<swss::FieldValueTuple>& values)
 {
+    MUTEX;
+
     SWSS_LOG_ENTER();
 
-    uint64_t value;
+    m_isDiscarded = false;
 
-    for (auto &counter : counterIds)
+    for (auto& fvt: values)
     {
-        sai_status_t status = m_vendorSai->getStats(
-                SAI_OBJECT_TYPE_INGRESS_PRIORITY_GROUP,
-                priorityGroupRid,
-                1,
-                (const sai_stat_id_t *)&counter,
-                &value);
+        auto& field = fvField(fvt);
+        auto& value = fvValue(fvt);
 
-        if (status != SAI_STATUS_SUCCESS)
-        {
-            SWSS_LOG_NOTICE("%s: counter %s is not supported on PG %s, rv: %s",
-                    m_instanceId.c_str(),
-                    sai_serialize_ingress_priority_group_stat(counter).c_str(),
-                    sai_serialize_object_id(priorityGroupRid).c_str(),
-                    sai_serialize_status(status).c_str());
+        auto shaStrings = swss::tokenize(value, ',');
 
-            continue;
+        if (field == POLL_INTERVAL_FIELD)
+        {
+            setPollInterval(stoi(value));
         }
-
-        if (m_statsMode == SAI_STATS_MODE_READ_AND_CLEAR)
+        else if (field == FLEX_COUNTER_STATUS_FIELD)
         {
-            status = m_vendorSai->clearStats(
-                    SAI_OBJECT_TYPE_INGRESS_PRIORITY_GROUP,
-                    priorityGroupRid,
-                    1,
-                    (const sai_stat_id_t *)&counter);
-
-            if (status != SAI_STATUS_SUCCESS)
-            {
-                SWSS_LOG_NOTICE("%s: clear counter %s is not supported on PG %s, rv: %s",
-                        m_instanceId.c_str(),
-                        sai_serialize_ingress_priority_group_stat(counter).c_str(),
-                        sai_serialize_object_id(priorityGroupRid).c_str(),
-                        sai_serialize_status(status).c_str());
-
-                continue;
-            }
+            setStatus(value);
+        }
+        else if (field == STATS_MODE_FIELD)
+        {
+            setStatsMode(value);
+        }
+        else if (field == QUEUE_PLUGIN_FIELD)
```

Comment:
I see there is a lot of similar logic here: to test the type of plugins/counters/etc. and handle them accordingly, like `createCounterContext`, `addCounter`, and `removeCounter` in `class FlexCounter`
maybe it's a good approach to have a mapping class, which maps the name to the corresponding methods/target strings
you may want to refer an example in qosorch.
I don't have a strong opinion. So I'm ok If you decide to go in this way.